test.py

```diff
@@ -19,41 +19,236 @@
+
+        try:
+            #Check if the test was already performed in a previous incomplete run
+            test_done_file_flag = os.path.join(output_path, 'done')
+            if os.path.exists(test_done_file_flag):
+                print(test_name,"already performed",sep=" ",end="\n\n\n")
+                outcomes.append(str(test_name)+" already performed")
+                continue
+            print("Starting",test_name,sep=" ")
+
+            if test_name == 'faas_predict':
+                # Build object
+                predictor_obj = Predictor(regressor_file=args.output+'/faas_test/LRRidge.pickle', output_folder=output_path, debug=args.debug)
+
+                # Perform prediction reading from a config file
+                predictor_obj.predict(config_file=configuration, mape_to_file=True)
+            else:
+                sequence_data_processor = sequence_data_processing.SequenceDataProcessing(configuration, debug=args.debug, output=output_path)
+                sequence_data_processor.process()
+        except Exception as e:
+            print("Exception",e,"raised", sep=' ')
+            outcomes.append(str(test_name)+" failed with exception "+str(e))
+        else:
+            outcomes.append(str(test_name)+" successfully run")
+        print('\n\n\n')
+
+    #Print results
+    print('\n\n\n\n\n\n\n\n\n\n\n\n*************Test Results*************')
+    i = 0
+    for outcome in outcomes:
+        i += 1
+        print(str(i)+')',outcome, sep=' ')
+
+    # Create success flag file
+    with open(done_file_flag, 'wb') as f:
+        pass
+
+
 
+def generate_tests():
+    tests = [
+        {
+            'Name': 'ernest',
+            'General':{
+                'run_num': 1,
+                'techniques':['LRRidge'],
+                'validation': 'All',
+                'hp_selection': 'All',
+                'y': 'y'
+            },
+            'DataPreparation':{
+                'input_path': 'inputs/simplest.csv',
+                'ernest': True,
+                'rename_columns': {"x1": "cores", "x2": "datasize"}
+            },
+            'LRRidge':{
+                'alpha': [0.1]
+            }
+        },
+        {
+            'Name': 'faas_test',
+            'General':{
+                'run_num': 1,
+                'techniques': ['LRRidge', 'DecisionTree'],
+                'hp_selection': 'KFold',
+                'validation': 'HoldOut',
+                'folds': 4,
+                'hold_out_ratio': 0.2,
+                'y': 'ave_response_time'
+            },
+            'DataPreparation':{
+                'input_path': 'inputs/faas_test.csv',
+                'inverse': ['Lambda'],
+                'product_max_degree': 2,
+                'product_interactions_only': True
+            },
+            'LRRidge':{
+                'alpha': [0.02, 0.1, 1.0]
+            },
+            'DecisionTree':{
+                'criterion': ['mse'],
+                'max_depth': [3],
```

Comment:
I would change `[3]` to `[2, 5]`, so that we always test at least two configurations for this model